tests/test_autonomy/test_cli/test_deploy/test_build/test_deployment.py

```diff
@@ -147,6 +147,56 @@ def test_docker_compose_build(
             ]
         )
 
+    def test_docker_compose_build_dev(
+        self,
+    ) -> None:
+        """Run tests."""
+
+        with mock.patch("os.chown"):
+            result = self.run_cli(
+                (
+                    self.service_id,
+                    str(self.keys_file),
+                    "--packages-dir",
+                    str(self.t / "packages"),
+                    "--o",
+                    str(self.t),
+                    "--force",
+                    "--dev",
+                )
+            )
+
+        build_dir = self.t / "abci_build"
+
+        assert result.exit_code == 0, f"{result.stdout_bytes}\n{result.stderr_bytes}"
+        assert build_dir.exists()
+
+        build_tree = list(map(lambda x: x.name, build_dir.iterdir()))
+        assert any(
+            [
+                child in build_tree
+                for child in ["persistent_storage", "nodes", "docker-compose.yaml"]
+            ]
```

Comment:
no need to revise, but still some needless list comprehensions, also a few lines below L187 and L191